torch_geometric/nn/models/graphmask_explainer.py

```diff
@@ -0,0 +1,838 @@
+            must lie between 0 and 10. (default: :obj:`5`)
+        lambda_optimizer_lr (float): The learning rate to optimize the
+            Lagrange multiplier. (default: :obj:`1e-2`)
+        init_lambda (float): The Lagrange multiplier. Value must lie between
+            0 and 1. (default: :obj:`0.55`)
+        allowance (float): A float value between 0 and 1 denotes tolerance
+            level. (default: :obj:`0.03`)
+        feat_mask_type (str, optional): Denotes the type of feature mask
+            that will be learned. Valid inputs are :obj:`"feature"` (a single
+            feature-level mask for all nodes), :obj:`"individual_feature"`
+            (individual feature-level masks for each node), and
+            :obj:`"scalar"` (scalar mask for each node).
+            (default: :obj:`"feature"`)
+        layer_type (str): The type of GNN layer being used in the GNN model.
+            (default: :obj:`GCN`)
+        allow_multiple_explanations: Switch to allow explainer to explain
+            node-level predictions for two or more nodes. Must set to
+            :obj:`False` while explaining graph-level predictions and
+            if only one node-level prediction is to be explained.
+            (default: :obj:`False`)
+        log (bool, optional): If set to :obj:`False`, will not log any
+            learning progress. (default: :obj:`True`)
+        **kwargs (optional): Additional hyper-parameters to override default
+            settings in
+            :attr:`~torch_geometric.nn.models.GraphMaskExplainer.coeffs`.
+    """
+
+    coeffs = {
+        'node_feat_size': 1.0,
+        'node_feat_reduction': 'mean',
+        'node_feat_ent': 0.1,
+    }
+
+    def __init__(self, num_layers, model_to_explain=None, epochs: int = 100,
+                 lr: float = 0.01, num_hops: Optional[int] = 3,
+                 return_type: str = 'log_prob', penalty_scaling: int = 5,
+                 lambda_optimizer_lr: int = 1e-2, init_lambda: int = 0.55,
+                 allowance: int = 0.03, feat_mask_type: str = 'scalar',
+                 layer_type: str = 'GCN',
+                 allow_multiple_explanations: bool = False, log: bool = True,
+                 **kwargs):
+        super().__init__(model_to_explain, lr, epochs, num_hops, return_type,
+                         log)
+        assert feat_mask_type in ['feature', 'individual_feature', 'scalar']
+        assert layer_type in ['GCN', 'GAT', 'FastRGCN']
+        assert 0 <= penalty_scaling <= 10
+        assert 0 <= init_lambda <= 1
+        assert 0 <= allowance <= 1
+
+        self.feat_mask_type = feat_mask_type
+        self.num_layers = num_layers
+        self.init_lambda = init_lambda
+        self.lambda_optimizer_lr = lambda_optimizer_lr
+        self.penalty_scaling = penalty_scaling
+        self.allowance = allowance
+        self.layer_type = layer_type
+        self.allow_multiple_explanations = allow_multiple_explanations
+        self.coeffs.update(kwargs)
```

Comment:
This is probably not a good idea, it will be shared across all instances of `GraphMaskExplainer`, so if a user ever created two with with different coeffs they will overwrite each other

Reply:
I've followed here the implementation of `GNNExplainer` which does this the same way as I did it here. Please check [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/gnn_explainer.py) on Line 77 for more details.

Reply:
I've followed here the implementation of `GNNExplainer` which does this the same way as I did it here. Please check [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/gnn_explainer.py) on Line 77 for more details.